packages/sitecore-jss-nextjs/src/components/NextImage.test.tsx

```diff
@@ -0,0 +1,232 @@
+/* eslint-disable no-unused-expressions */
+import chai from 'chai';
+import chaiString from 'chai-string';
+import { mount } from 'enzyme';
+import React from 'react';
+import { imageField as eeImageData } from '../testData/ee-data';
+import { NextImage, ImageField, loader } from './NextImage';
+
+const expect = chai.use(chaiString).expect;
+
+describe('<NextImage />', () => {
+  describe('Next loader function', () => {
+    const props = {
+      src: '/assets/img/test0.png',
+      width: 100,
+      height: 50,
+    };
+    const result = loader(props);
+
+    it('should return url and set it to src', () => {
+      expect(result).to.be.a('string');
+      expect(result).to.equal('https://cm.jss.localhost/assets/img/test0.png?mw=100');
+    });
+  });
+
+  // TODO: test error cases
+  describe('error cases', () => {
+    const props = {
+      src: '/assets/img/test0.png',
+      loader: () => new URL('test').href,
+    };
+
+    it('should throw an error if src is present', () => {
+      expect(() => mount(<NextImage {...props} />)).throws(
+        'Detected conflicting props src or loader. If you wish to use these props, use next/image directly.'
+      );
+    });
+
+    it('should throw an error if loader is present', () => {
```

Comment:
These error case tests should use isolated `props` with only the prop they are testing (`src` or `loader`).
Also, add test for `srcSet` error case.

Reply:
I have remove the srcSet test because Nextjs makes sure to throw an error if the user passes the srcSet prop.

Reply:
If that's the case (next/image is throwing an error already when using `srcSet`), then yes - we don't need a test and should get rid of [our check](https://github.com/Sitecore/jss/pull/909/files#diff-94787a2491b63b1167cab2caf9afa18626d4ca37c062884aa705fc8e7ded343eR53-R57).